polkadot/node/network/collator-protocol/src/collator_side/metrics.rs

```diff
@@ -234,140 +234,152 @@ pub(crate) const MAX_AVAILABILITY_DELAY: BlockNumber = 10;
+			gum::debug!(
+				target: crate::LOG_TARGET_STATS,
+				?head,
+				"Collation already backed in a fork, skipping",
+			);
+			return
+		}
 
+		entry.set_backed_at(block_number);
+		if let Some(latency) = entry.backed() {
 			// Observe the backing latency since the collation was fetched.
 			let maybe_latency =
 				entry.backed_latency_metric.take().map(|metric| metric.stop_and_record());
-
 			gum::debug!(
 				target: crate::LOG_TARGET_STATS,
-				latency_blocks = ?entry.backed(),
+				latency_blocks = ?latency,
 				latency_time = ?maybe_latency,
 				relay_block = ?leaf,
-				?relay_parent,
-				?para_id,
+				relay_parent = ?entry.relay_parent,
+				para_id = ?receipt.descriptor.para_id(),
 				?head,
 				"A fetched collation was backed on relay chain",
 			);
-
-			metrics.on_collation_backed(
-				(block_number.saturating_sub(entry.relay_parent_number)) as f64,
-			);
-
-			entry
-		})
+		}
 	}
 
-	// Mark a previously backed collation as included and return the stats.
-	// After this call, the collation is no longer trackable.
+	// Mark a previously backed collation as included.
 	//
-	// Block built on top of N is earliest backed at N + 1.
-	// Returns `None` if there collation is not in tracker.
+	// Block built on top of N is earliest included at N + 2.
 	pub fn collation_included(
 		&mut self,
 		block_number: BlockNumber,
 		leaf: H256,
 		receipt: CandidateReceipt,
-		metrics: &Metrics,
-	) -> Option<CollationStats> {
+	) {
 		let head = receipt.descriptor.para_head();
+		let Some(entry) = self.entries.get_mut(&head) else {
+			gum::debug!(
+				target: crate::LOG_TARGET_STATS,
+				?head,
+				"Included collation not found in tracker",
+			);
+			return;
+		};
 
-		self.entries.remove(&head).map(|mut entry| {
-			entry.included_at = Some(block_number);
-
-			if let Some(latency) = entry.included() {
-				metrics.on_collation_included(latency as f64);
-
-				let para_id = receipt.descriptor.para_id();
-				let relay_parent = receipt.descriptor.relay_parent();
-
-				gum::debug!(
-					target: crate::LOG_TARGET_STATS,
-					?latency,
-					relay_block = ?leaf,
-					?relay_parent,
-					?para_id,
-					head = ?receipt.descriptor.para_head(),
-					"Collation included on relay chain",
-				);
-			}
+		if entry.included().is_some() {
+			gum::debug!(
+				target: crate::LOG_TARGET_STATS,
+				?head,
+				"Collation already included in a fork, skipping",
+			);
+			return
+		}
 
-			entry
-		})
+		entry.set_included_at(block_number);
+		if let Some(latency) = entry.included() {
+			gum::debug!(
+				target: crate::LOG_TARGET_STATS,
+				?latency,
+				relay_block = ?leaf,
+				relay_parent = ?entry.relay_parent,
+				para_id = ?receipt.descriptor.para_id(),
+				head = ?receipt.descriptor.para_head(),
+				"Collation included on relay chain",
+			);
+		}
 	}
 
 	// Returns all the collations that have expired at `block_number`.
 	pub fn drain_expired(&mut self, block_number: BlockNumber) -> Vec<CollationStats> {
```

Comment:
not your code, but the doc comment `//` should be `///` to be proper doc, right?